sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -124,7 +124,14 @@ case class SortArray(base: Expression, ascendingOrder: Expression)
 
   override def checkInputDataTypes(): TypeCheckResult = base.dataType match {
     case ArrayType(dt, _) if RowOrdering.isOrderable(dt) =>
-      TypeCheckResult.TypeCheckSuccess
+      ascendingOrder match {
+        case Literal(_: Boolean, BooleanType) =>
+          TypeCheckResult.TypeCheckSuccess
+        case _ =>
+          TypeCheckResult.TypeCheckFailure(
+            s"Sort order in second argument requires ${BooleanType.simpleString} type, however," +
```

Comment:
Can you explicitly mention that the error is in context of `sort_array()` ? For SQL query which spans 100+ lines, its easy to get lost while tracking why it failed. Having better error messages would make that easier for users.